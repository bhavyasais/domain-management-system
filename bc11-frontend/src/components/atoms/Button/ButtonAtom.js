import React from "react";
import Button from "@material-ui/core/Button";
import { makeStyles, Typography } from "@material-ui/core";
import PropTypes from "prop-types";
import clsx from "clsx";
const useStyles = makeStyles({
  buttonLabel: {
    overflow: "hidden",
    textOverflow: "ellipsis",
  },
});

export default function ButtonAtom({
  value,
  variant,
  disabled,
  color,
  onClick,
  textColor,
  rootStyle,
  labelStyle,
  textVariant,
  maxWidth,
  ...restProps
}) {
  const classes = useStyles();
  return (
    <div>
      <Button
        variant={variant}
        disabled={disabled}
        style={{ backgroundColor: color, maxWidth: maxWidth }}
        onClick={onClick}
        classes={{
          root: rootStyle,
          label: clsx(labelStyle, classes.buttonLabel),
        }}
        {...restProps}
      >
        <Typography variant={textVariant} className={classes.buttonLabel}>
          {value}
        </Typography>
      </Button>
    </div>
  );
}
Button.propTypes = {
  value: PropTypes.string,
  color: PropTypes.string,
  variant: PropTypes.string,
};
Button.defaultProps = {
  variant: "contained",
};
