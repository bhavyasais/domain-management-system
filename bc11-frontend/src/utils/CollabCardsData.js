import box from "../assets/atoms_icons_box.svg";
import gsuite from "../assets/g-suite.jpg";
import okta from "../assets/atoms_icons_okta.svg";
export const collabCardsData1 = [
  {
    title: "box card",
    image: box ,
    component: "img",
    height: "200",
    name: "Box",
  },
  {
    title: "g suite card",
    image:  gsuite ,
    component: "img",
    height: "200",
    name: "G Suite",
  },
  {
    title: "docu sign card",
    image:
      "https://documentautomation.net/wp-content/uploads/2016/05/docusign-circle-large.png",
    component: "img",
    height: "200",
    name: "Docusign",
  },
  {
    title: "figma card",
    image:
      "https://docs.microsoft.com/en-us/windows/uwp/design/downloads/images/figma.png",
    component: "img",
    height: "200",
    name: "Figma",
  },
];
export const collabCardsData2 = [
    {
      title: "okta card",
      image:  okta ,
      component: "img",
      height: "200",
      name: "Okta",
    },
    {
      title: "slack card",
      image: "https://cdn.mos.cms.futurecdn.net/SDDw7CnuoUGax6x9mTo7dd.jpg",
      component: "img",
      height: "200",
      name: "Slack",
    },
  ];
  
