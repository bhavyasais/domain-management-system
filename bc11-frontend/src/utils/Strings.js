export const TRUSTGROUP_COLORS = {
  'Top 100': '#daefa9',
  'Top 10000': '#b0f1f0',
  'Top 30000': '#fdd570',
  'Do not trust': '#ffada5',
};

export const TRUST_GROUPS = ['Top 100', 'Top 1000', 'Top 3000', 'Do not trust'];

export const RELATIONSHIPS = ['Customer', 'Vendor', 'Partner'];

export const RELATIONS = [
  { groupName: 'Customer' },
  { groupName: 'Vendor' },
  { groupName: 'Partner' },
];

export const RELATIONSHIP_COLORS = {
  Customer: '#fdd570',
  Partner: '#b0f1f0',
  Vendor: '#daefa9',
};

export const STATUS = {
  SUCCESS: 'SUCCESS',
  FAILURE: 'FAILURE',
};

export const RELATIONSHIP_CARD_LEGEN_1 = ' Lorem ipsum dolor sit amet, consectetur adipiscing elit';
export const RELATIONSHIP_CARD_LEGEN_2 = ' Lorem ipsum dolor sit amet, consectetur adipiscing elit';
export const DASHBOARD_HEADER_LABEL = 'Dashboard';
export const GREET_USER = 'Good morning,';
export const SUBHEADING = 'Here is whatâ€™s happening with your domains';
export const GRAPH_CARDS = ['Total Domains', 'Total Received', 'Trusted', 'Do Not Trust'];
export const PLACEHOLDER = 'Mail@gmail.com';
export const DESCRIPTION =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat';
export const SOURCE_DETAILS_HEADER = 'Source Details';
export const THREAT_DETAILS_HEADER = 'Threat Details';
