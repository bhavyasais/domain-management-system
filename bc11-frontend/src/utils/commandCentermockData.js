import React from "react";
import faker from "faker";

const clickedCheckbox = (params) => {};

const checkboxRenderer = (params) => {
  return (
    <div>
      <input type="checkbox" onClick={() => clickedCheckbox(params)} />
    </div>
  );
};

const headerRenderer = (params) => {
  const name = params.displayName;
  return (
    <div>
      <input type="checkbox" onClick={() => clickedCheckbox(params)} />
      <label>{name}</label>
    </div>
  );
};

const redoFunction = (headerName, headerField) => {
  return {
    headerName: headerName,
    field: headerField,
    cellRendererFramework: (params) => {
      return checkboxRenderer(params);
    },
    headerComponentFramework: (params) => {
      return headerRenderer(params);
    },
    cellStyle: { textAlign: "center" },
  };
};

const columnDefs = [
  { headerName: "S no", valueGetter: "node.rowIndex + 1" },
  {
    headerName: "A Trust Group",
    field: "trustGroup",
    sortable: true,
    filter: true,
  },
];

columnDefs.push(redoFunction("Edit File?", "editFile"));
columnDefs.push(redoFunction("View File?", "viewFile"));
columnDefs.push(redoFunction("Delete File?", "deleteFile"));
columnDefs.push(redoFunction("Share File?", "shareFile"));

export const trustGroupcolumnDefs = columnDefs;

export const trustGrouprowData = [
  {
    trustGroup: "DO NOT TRUST",
    editFile: true,
    viewFile: false,
    deleteFile: false,
    shareFile: false,
  },
  {
    trustGroup: "Top 10",
    editFile: false,
    viewFile: false,
    deleteFile: true,
    shareFile: false,
  },
  {
    trustGroup: "Top 10000",
    editFile: false,
    viewFile: false,
    deleteFile: false,
    shareFile: false,
  },
  {
    trustGroup: "Top 30000",
    editFile: false,
    viewFile: false,
    deleteFile: false,
    shareFile: false,
  },
];

export const DomaincolumnDefs = [
  { headerName: "S no", valueGetter: "node.rowIndex + 1" },
  { headerName: "A Domain", field: "domain", sortable: true, filter: true },
  {
    headerName: "Trust Score",
    field: "trustScore",
    sortable: true,
    filter: true,
  },
  {
    headerName: "Trust Group",
    field: "trustGroup",
    sortable: true,
    filter: true,
  },
];

export const DomainrowData = [
  {
    domain: faker.lorem.word(),
    email: "example@yahoo.com",
    trustScore: faker.random.number(),
    trustGroup: "Do not trust",
    id: 0,
  },
  {
    domain: faker.lorem.word(),
    email: "user123@gamil.com", //NOSONAR
    trustScore: faker.random.number(),
    trustGroup: "Top 100",
    id: 1,
  },
  {
    domain: faker.lorem.word(),
    email: "testEmail@Zemoso.com", //NOSONAR
    trustScore: faker.random.number(),
    trustGroup: "Do not trust",
    id: 2,
  },
  {
    domain: faker.lorem.word(),
    email: "testUser@Zemoso.com", //NOSONAR
    trustScore: faker.random.number(),
    trustGroup: "Top 1000",
    id: 3,
  },
  {
    domain: faker.lorem.word(),
    email: "testPerson@Zemoso.com", //NOSONAR
    trustScore: faker.random.number(),
    trustGroup: "Top 3000",
    id: 4,
  },
];
export const RelationshipData = [
  {
    domain: faker.lorem.word(),
    trustScore: faker.random.number(),
    relationship: "Partner",
  },
  {
    domain: faker.lorem.word(),
    trustScore: faker.random.number(),
    relationship: "Vendor",
  },
  {
    domain: faker.lorem.word(),
    trustScore: faker.random.number(),
    relationship: "Customer",
  },
  {
    domain: faker.lorem.word(),
    rustScore: faker.random.number(),
    relationship: "Partner",
  },
  {
    domain: faker.lorem.word(),
    trustScore: faker.random.number(),
    relationship: "Vendor",
  },
];
