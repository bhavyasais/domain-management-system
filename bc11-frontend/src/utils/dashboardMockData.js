import React from 'react';

export const graphCardDetails = [
  {
    cardName: 'Total Domains',
    totalCount: 30,
    additionalCount: 1,
  },
  {
    cardName: 'Total Received',
    totalCount: 30,
    additionalCount: 5,
  },
  {
    cardName: 'Trusted',
    totalCount: 30,
    additionalCount: 5,
  },
  {
    cardName: 'Do Not Trust',
    totalCount: 30,
    additionalCount: 1,
  },
];

export const graphData = {
  value: 'zemoso',
  name: 'zemoso',
  data: [
    { date: 'Mon', value: 30 },
    { date: 'Tue', value: 15 },
    { date: 'Wen', value: 12 },
    { date: 'Thu', value: 40 },
    { date: 'Fri', value: 20 },
    { date: 'Sat', value: 5 },
    { date: 'Sun', value: 30 },
  ],
};

export const trustGroupGraphData = [
  {
    value: 'zemoso',
    name: 'zemoso',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 5 },
      { date: '3rd Dec', value: 20 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 12 },
      { date: '6th Dec', value: 25 },
      { date: '7th Dec', value: 30 },
    ],
  },
  {
    value: 'stanford',
    name: 'stanford',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 15 },
      { date: '3rd Dec', value: 12 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 20 },
      { date: '6th Dec', value: 5 },
      { date: '7th Dec', value: 30 },
    ],
  },
];

export const trustScoreGraphData = [
  {
    value: 'zemoso',
    name: 'zemoso',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 5 },
      { date: '3rd Dec', value: 20 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 12 },
      { date: '6th Dec', value: 25 },
      { date: '7th Dec', value: 30 },
      { date: '8th Dec', value: 40 },
      { date: '9th Dec', value: 12 },
      { date: '10th Dec', value: 25 },
      { date: '11th Dec', value: 30 },
      { date: '12th Dec', value: 25 },
      { date: '13th Dec', value: 30 },
      { date: '14th Dec', value: 25 },
      { date: '15th Dec', value: 30 },
      { date: '16th Dec', value: 25 },
      { date: '17th Dec', value: 30 },
    ],
  },
  {
    value: 'stanford',
    name: 'stanford',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 5 },
      { date: '3rd Dec', value: 20 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 12 },
      { date: '6th Dec', value: 25 },
      { date: '7th Dec', value: 30 },
      { date: '8th Dec', value: 40 },
      { date: '9th Dec', value: 12 },
      { date: '10th Dec', value: 25 },
      { date: '11th Dec', value: 30 },
      { date: '12th Dec', value: 25 },
      { date: '13th Dec', value: 30 },
      { date: '14th Dec', value: 25 },
      { date: '15th Dec', value: 30 },
      { date: '16th Dec', value: 25 },
      { date: '17th Dec', value: 30 },
    ],
  },
];
export const trustTabData = [
  {
    value: 'zemoso',
    name: 'zemoso',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 5 },
      { date: '3rd Dec', value: 20 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 12 },
      { date: '6th Dec', value: 25 },
      { date: '7th Dec', value: 30 },
    ],
  },
  {
    value: 'stanford',
    name: 'stanford',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 15 },
      { date: '3rd Dec', value: 12 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 20 },
      { date: '6th Dec', value: 5 },
      { date: '7th Dec', value: 30 },
    ],
  },
];
export const doNotTrustTabData = [
  {
    value: 'zemoso',
    name: 'zemoso',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 5 },
      { date: '3rd Dec', value: 20 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 12 },
      { date: '6th Dec', value: 25 },
      { date: '7th Dec', value: 30 },
    ],
    color: '#ff3162',
  },
  {
    value: 'stanford',
    name: 'stanford',
    data: [
      { date: 'Mon', value: 30 },
      { date: 'Tue', value: 15 },
      { date: 'Wen', value: 12 },
      { date: 'Thu', value: 40 },
      { date: 'Fri', value: 20 },
      { date: 'Sat', value: 5 },
      { date: 'Sun', value: 30 },
    ],
    color: '#ff3162',
  },
];
export const domains = [
  { name: 'zemosolabs', value: 'zemosolabs' },
  { name: 'stanford', value: 'zemostanfordsolabs' },
];
