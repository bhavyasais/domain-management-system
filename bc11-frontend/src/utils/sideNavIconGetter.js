import Graph from '../assets/atoms_icon_graph.svg';
import Pie from '../assets/atoms_icon_pie.svg';
import Sheet from '../assets/atoms_icon_excel.svg';
import Mail from '../assets/mail.png';
const sideNavIcons = [
  { icon: Pie, active: false },
  { icon: Graph, active: true },
  { icon: Mail, active: true },
  { icon: Sheet, active: false },
];

const getSideBarIcons = () => sideNavIcons;

const sideNavIconGetter = {
  getSideBarIcons,
};

export default sideNavIconGetter;
