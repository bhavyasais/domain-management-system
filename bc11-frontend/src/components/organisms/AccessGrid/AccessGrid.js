import React, { useCallback, useEffect, useState } from 'react';
import { withStyles } from '@material-ui/core';
import { AgGridReact, AgGridColumn } from 'ag-grid-react';
import PropTypes from 'prop-types';
import AddIcon from '@material-ui/icons/Add';
import Button from '../../atoms/Button/ButtonAtom';

import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
import TrustGroupService from '../../../services/TrustGroupService';
import {
  CreateTrustGroupPermissionArray,
  createUpdateTrustGroupParam,
  getPermission,
} from '../../../utils/helper';
import DomainService from '../../../services/DomainService';
import GoogleSuiteService from '../../../services/GoogleSuiteService';

const styles = () => ({
  addNewRow: {
    height: '36px',
    width: '690',
    border: '1px solid #cccccc',
    borderTop: '0',
    display: 'flex',
    alignItems: 'center',
    paddingLeft: '8px',
    backgroundColor: '#ffffff',
  },
  buttonLabelStyle: {
    color: '#1d4cd7',
    textTransform: 'none',
  },
});

const AccessGrid = ({
  classes,
  setSelectedRows,
  index,
  setTotalRows,
  trustGroups,
  deptId,
  addTrustGroup,
  updateTrustGroup,
}) => {
  const [gridApi, setGridApi] = useState(null); // NOSONAR
  const [gridColumnApi, setGridColumnApi] = useState(null); // NOSONAR
  const [list, setList] = useState([]);

  const totalHeigh = list.length * 40 + 51.4;

  const onGridReady = (params) => {
    setGridApi(params.api);
    setGridColumnApi(params.columnApi);
  };

  const updateAccess = async (item, colId, checked) => {
    const params = createUpdateTrustGroupParam(item, colId, checked);

    params.collaborators_id = localStorage.getItem('Collaboration');
    params.departments_id = JSON.parse(localStorage.getItem('dept')).id;

    const googleObj = {
      departmentName: JSON.parse(localStorage.getItem('dept')).name,
      type: 'domain',
      permissionType: localStorage.getItem(item.trustGroupName),
    };

    try {
      await TrustGroupService.updateTrustGroupAccess(params);

      const domains = await DomainService.getAllDomainsForDepartment({
        collaborators_id: localStorage.getItem('Collaboration'),
        departments_id: JSON.parse(localStorage.getItem('dept')).id,
      });

      googleObj.clients = domains
        .filter((domainObj) => domainObj.trustGroupName === item.trustGroupName)
        .map((obj) => obj.domain_name);
      console.log('google,', googleObj);

      await GoogleSuiteService.changePermission(googleObj);
    } catch (error) {
      console.warn('Error in updating trustgroup Access : ', error);
    }
  };

  const clickedCheckbox = (e, params) => {
    const { checked } = e.target;
    const { colId } = params.column;
    params.node.setDataValue(colId, checked);
    const newAarray = [...params.list];
    newAarray[params.rowIndex] = params.data;
    getPermission(newAarray);
    setList(newAarray);
    updateAccess(params.data, colId, checked);
  };

  const checkboxRenderer = (params) => {
    return (
      <div>
        <input type="checkbox" onClick={(e) => clickedCheckbox(e, params)} checked={params.value} />
      </div>
    );
  };

  const headerRenderer = (params) => {
    const name = params.displayName;
    return (
      <div>
        <input type="checkbox" onClick={() => clickedCheckbox(params)} />
        <label>{name}</label>
      </div>
    );
  };

  const fetchAllPermissions = useCallback(async () => {
    try {
      const result = await TrustGroupService.getAllTrustGroupsWithPermissions({
        collaborators_id: localStorage.getItem('Collaboration'),
        departments_id: deptId,
      });
      const permissionArray = CreateTrustGroupPermissionArray(result, trustGroups); // This should ideally be handled by backend
      setList([...permissionArray]);
    } catch (error) {
      console.warn('Error in fetch Access permissions : ', error);
    }
  }, [trustGroups, deptId]);

  useEffect(() => {
    fetchAllPermissions();
  }, [fetchAllPermissions]);

  useEffect(() => {
    gridApi?.refreshCells();
  }, [list]);

  useEffect(() => {
    setTotalRows((prevArray) => {
      const newArray = [...prevArray];
      newArray[index] = list.length;
      return newArray;
    });
  }, [list, index]);

  const handleOnSelectionChanged = () => {
    const selected = gridApi.getSelectedRows().length;
    setSelectedRows((prevArrray) => {
      const newArray = [...prevArrray];
      newArray[index] = selected;
      return newArray;
    });
  };

  function createNewTrustGroup() {
    let groupName = 'Top 10';
    while (trustGroups.find((item) => item.groupName === groupName)) {
      groupName += '0';
    }
    return groupName;
  }

  const addNewTrustGroup = () => {
    const newItems = createNewTrustGroup();
    addTrustGroup(newItems);
  };

  const handleCellEditingStopped = (param) => {
    if (param.oldValue !== param.newValue) {
      updateTrustGroup(param.data);
    }
  };

  return (
    <div style={{ width: '100%', height: '100%', flex: 1 }}>
      <div
        style={{
          height: list?.length !== 0 ? totalHeigh : '360px',
          width: '1091px',
        }}
        className="ag-theme-alpine"
      >
        <AgGridReact
          defaultColDef={{
            resizable: true,
            sortable: true,
            filter: true,
          }}
          rowSelection="multiple"
          rowDragManaged
          getRowHeight={() => 40}
          onGridReady={onGridReady}
          rowData={list}
          onSelectionChanged={handleOnSelectionChanged}
          onSortChanged={() => {
            gridApi.refreshCells();
          }}
          onRowDragEnd={() => {
            gridApi.refreshCells();
          }}
          onCellEditingStopped={handleCellEditingStopped}
        >
          <AgGridColumn
            headerName="S no"
            valueGetter="node.rowIndex + 1"
            width={78}
            lockPosition="true"
            rowDrag
          />
          <AgGridColumn headerName="A Trust Group" field="trustGroupName" editable />
          <AgGridColumn
            headerName="Edit File?"
            field="editFile"
            cellRendererFramework={checkboxRenderer}
            cellRendererParams={{
              list,
            }}
            headerComponentFramework={headerRenderer}
          />
          <AgGridColumn
            headerName="View File?"
            field="viewFile"
            cellRendererFramework={checkboxRenderer}
            cellRendererParams={{
              list,
            }}
            headerComponentFramework={headerRenderer}
          />
          <AgGridColumn
            headerName="Comment on File?"
            field="commentFile"
            cellRendererFramework={checkboxRenderer}
            cellRendererParams={{
              list,
            }}
            headerComponentFramework={headerRenderer}
          />
          <AgGridColumn
            headerName="Transfer ownership of File?"
            field="transferFile"
            cellRendererFramework={checkboxRenderer}
            cellRendererParams={{
              list,
            }}
            headerComponentFramework={headerRenderer}
          />
        </AgGridReact>
        <div className={classes.addNewRow}>
          <Button
            variant="text"
            className={classes.button}
            startIcon={<AddIcon />}
            disableRipple
            value="New row"
            labelStyle={classes.buttonLabelStyle}
            onClick={addNewTrustGroup}
            textVariant="caption"
          />
        </div>
      </div>
    </div>
  );
};

AccessGrid.propTypes = {
  setSelectedRows: PropTypes.func,
  index: PropTypes.number.isRequired,
  setTotalRows: PropTypes.func,
  trustGroups: PropTypes.arrayOf(PropTypes.string),
};
AccessGrid.defaultProps = {
  setSelectedRows: () => {},
  setTotalRows: () => {},
  trustGroups: [],
};

export default withStyles(styles)(AccessGrid);
