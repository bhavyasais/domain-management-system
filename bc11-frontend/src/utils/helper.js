import { RELATIONSHIP_COLORS, TRUSTGROUP_COLORS } from './Strings';

export const FixDomainRowTrustGroup = (array) =>
  array.map((row, idx) => ({
    ...row,
    trustGroup: {
      backgroundColor: TRUSTGROUP_COLORS[row?.trustGroup],
      label: row?.trustGroup,
      idx,
    },
  }));

export const UpdateTrustGroups = (array) =>
  array.map((value) => ({
    backgroundColor: TRUSTGROUP_COLORS[value.name] ?? '#eeeeee',
    label: value.name,
  }));

export const UpdateDomainsValue = (array, data) => {
  const newData = array.map((row) => {
    return row.id === data.id ? { ...data } : { ...row };
  });
  return newData;
};

export const FixDomainRow = (array) =>
  array.map((row) => ({
    ...row,
    relationship: {
      backgroundColor: RELATIONSHIP_COLORS[row?.relationship],
      label: row?.relationship,
    },
  }));

export const CreateTrustGroupPermissionArray = (array, trustgroups) => {
  const newArray = Array.from({ length: trustgroups.length }, () => ({}));

  trustgroups.forEach((obj, idx) => {
    newArray[idx].trustGroupName = obj.groupName;
    newArray[idx].trust_groups_id = obj.id;
    newArray[idx].viewFile = false;
    newArray[idx].commentFile = false;
    newArray[idx].editFile = false;
    newArray[idx].transferFile = false;
  });

  const finalArray = newArray.map((obj) => {
    const newObj = obj;
    const matchedObjects = array.filter((item) => item.trust_group_name === obj.trustGroupName);
    matchedObjects.forEach((item) => {
      if (item.permission_type === 'READ') {
        newObj.viewFile = item.is_active;
        newObj.viewFileId = item.id;
        newObj.viewFileAccessId = item.access_permission_types_id;
      } else if (item.permission_type === 'COMMENT') {
        newObj.commentFile = item.is_active;
        newObj.commentFileId = item.id;
        newObj.commentFileAccessId = item.access_permission_types_id;
      } else if (item.permission_type === 'EDIT') {
        newObj.editFile = item.is_active;
        newObj.editFileId = item.id;
        newObj.editFileAccessId = item.access_permission_types_id;
      } else if (item.permission_type === 'TRANSFER OWNERSHIP') {
        newObj.transferFile = item.is_active;
        newObj.transferFileId = item.id;
        newObj.transferFileAccessId = item.access_permission_types_id;
      }
    });
    return newObj;
  });
  return finalArray;
};

export const createUpdateTrustGroupParam = (item, colId, checked) => {
  const newObj = {};
  newObj.is_active = checked;
  newObj.trust_groups_id = item.trust_groups_id;

  if (colId === 'viewFile') {
    newObj.id = item.viewFileId;
    newObj.access_permission_types_id = item.viewFileAccessId;
  } else if (colId === 'commentFile') {
    newObj.id = item.commentFileId;
    newObj.access_permission_types_id = item.commentFileAccessId;
  } else if (colId === 'editFile') {
    newObj.id = item.editFileId;
    newObj.access_permission_types_id = item.editFileAccessId;
  } else if (colId === 'transferFile') {
    newObj.id = item.transferFileId;
    newObj.access_permission_types_id = item.transferFileAccessId;
  }
  return newObj;
};

export const getPermission = (array) => {
  array.forEach((item) => {
    if (item?.transferFile) {
      localStorage.setItem(item.trustGroupName, 'writer');
    } else if (item?.editFile) {
      localStorage.setItem(item.trustGroupName, 'writer');
    } else if (item?.commentFile) {
      localStorage.setItem(item.trustGroupName, 'commenter');
    } else if (item?.viewFile) {
      localStorage.setItem(item.trustGroupName, 'reader');
    } else {
      localStorage.setItem(item.trustGroupName, 'none');
    }
  });
};

export const createNewDomainName=(list) =>{
  let domain_name = 'New Domain';
  while (list?.find((item) => item?.domain_name === domain_name)) {
    domain_name = domain_name.concat(" ",list.length);
  }
  return domain_name;
}

export const validateDomainName=(list, node, gridApi, gridColumnApi) =>{
  if (node?.column?.colId === 'domain_name') {
    if (list?.find((item) => item?.domain_name?.trim() === node?.data?.domain_name?.trim())) {
      gridApi.startEditingCell({
        rowIndex: node?.rowIndex,
        colKey:gridColumnApi?.getAllColumns()?.[1]?.colId,
      });
    }
  }
}

export const getDomainCount=(list, node)=> {
  var count = 0;
  for (var i = 0; i < list.length; i++) {
    if (list?.[i]?.domain_name.trim() === node?.oldValue.trim()){ count++}
  }
  return count;
}
