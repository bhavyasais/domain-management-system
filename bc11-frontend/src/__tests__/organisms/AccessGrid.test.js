import React from 'react';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { TRUST_GROUPS } from '../../utils/Strings';
import ButtonAtom from '../../components/atoms/Button/ButtonAtom';
import { AgGridReact } from 'ag-grid-react/lib/agGridReact';
import AccessGrid from '../../components/organisms/AccessGrid/AccessGrid';
describe('AccessGrid', () => {
  var component;
  const jestFunction = jest.fn();
  beforeEach(() => {
    act(() => {
      component = mount(
        <AccessGrid
          onClick={jestFunction}
          trustGroups={TRUST_GROUPS}
          setSelectedRows={jestFunction}
          setTotalRows={jestFunction}
          deptID={0}
          index={0}
        />,
      );
    });
  });
  it('should render correctly', () => {
    let wrapper;
    wrapper = renderer.create(
      <AccessGrid
        trustGroups={TRUST_GROUPS}
        setSelectedRows={() => {}}
        setTotalRows={() => {}}
        deptID={0}
        index={0}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
    it('ag grid react test', () => {
      component.find(AgGridReact).at(0).props().onGridReady({params:{api:"abc"}});
      component.find(AgGridReact).at(0).props().getRowHeight(()=>{});
    });
});
