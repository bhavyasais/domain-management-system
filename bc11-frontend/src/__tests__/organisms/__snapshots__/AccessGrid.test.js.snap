// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessGrid should render correctly 1`] = `
<div
  style={
    Object {
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    className="ag-theme-alpine"
    style={
      Object {
        "height": "360px",
        "width": "1091px",
      }
    }
  >
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
    <div
      className="AccessGrid-addNewRow-1"
    >
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "backgroundColor": undefined,
              "maxWidth": undefined,
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label AccessGrid-buttonLabelStyle-2 makeStyles-buttonLabel-3"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            <span
              className="MuiTypography-root makeStyles-buttonLabel-3 MuiTypography-caption"
            >
              New row
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
